sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -191,6 +191,31 @@ object Cast {
   }
 
   def resolvableNullability(from: Boolean, to: Boolean): Boolean = !from || to
+
+  def processFloatingPointSpecialLiterals(v: String, isFloat: Boolean): Any = {
+    val str = v.trim
+    if (str.equalsIgnoreCase("infinity") || str.equalsIgnoreCase("+infinity")) {
```

Comment:
Do you need Locale.ROOT here? it's a really obscure possibility, but "i" can cause problems

Reply:
@srowen OK.. will add Locale.ROOT by following how its used in other places.